src/edu/stanford/dlss/wowza/SulWowza.java

```diff
@@ -15,73 +17,203 @@
+
+    /** require streamName to have format "aa/000/aa/0000/sample.mp4"  (druid tree)/(media_fname) */
+    boolean validateStreamName(String streamName)
     {
-        getLogger().info("Tommy: SulWowza token = " + token);
-        return true;
+        if (streamName == null || streamName.length() <= DRUID_TREE_LENGTH)
+        {
+            getLogger().error(this.getClass().getSimpleName() + ": streamName missing or implausibly short" +
+                                (streamName == null ? "" : ": " + streamName));
+            return false;
+        }
+        String[] pieces = streamName.split("/");
+        if (pieces.length == 5)
+            return true;
+        else
+        {
+            getLogger().error(this.getClass().getSimpleName() + ": unable to parse druid and filename from streamName " + streamName);
+            return false;
+        }
+    }
+
+    static private final Pattern DRUID_REGEX_PATTERN = Pattern.compile("[a-z]{2}\\d{3}[a-z]{2}\\d{4}");
+
+    /** the druid of the media file to be streamed;  needed to verify stacks token
+     * Assumption:  validateStreamName() was already called */
+    String getDruid(String streamName)
+    {
+        String[] pieces = streamName.split("/");
+        String druid = pieces[0] + pieces[1] + pieces[2] + pieces[3];
+        if (DRUID_REGEX_PATTERN.matcher(druid).matches())
+            return druid;
+        else
+            return null;
+    }
+
+    /** the media file to be streamed, as stored under the druid-tree;  needed to verify stacks token
+     * Assumption:  validateStreamName() was already called */
+    String getFilename(String streamName)
+    {
+        String filename = streamName.substring(streamName.lastIndexOf('/') + 1);
+        if (filename.length() > 0)
+            return filename;
+        else
+            return null;
+    }
+
+    /** Assumption: stacksToken, druid, userIp and filename are all reasonable values (non-null, not empty, etc.) */
+    String getVerifyStacksTokenUrl(String stacksToken, String druid, String filename, String userIp)
+    {
+        // TODO:  Url encode anything?   utf-8 charset affect anything? (filename, stacksToken)
+        String queryStr = "stacks_token=" + stacksToken + "&user_ip=" + userIp;
+        String fullUrl = stacksTokenVerificationBaseUrl + "/media/" + druid + "/" + filename + "/verify_token?" + queryStr;
+        try
+        {
+            new URL(fullUrl);
+            return fullUrl;
+        }
+        catch (MalformedURLException err)
+        {
+            getLogger().error(this.getClass().getSimpleName() + " bad URL for stacks_token verification: ", err);
+            return null;
+        }
+    }
+
+    /** Assumption: verifyStacksTokenRequestUrl is a valid URL */
+    boolean verifyStacksToken(String verifyStacksTokenUrl)
+    {
+        try
+        {
+            URL tokenVerifyURL = new URL(verifyStacksTokenUrl);
+            URLConnection stacksConn = tokenVerifyURL.openConnection();
```

Comment:
I *think* you can say

HttpURLConnection stacksConn = tokenVerifyURL.openConnection()

instead here. Just to be explicit about what type you're using and to avoid the cast below.